### PR TITLE
Rhys/partial silencing2

### PR DESCRIPTION
**Pull request by wheresrhys**
@alicebartlett @AlbertoElias @onishiweb 

Changes compared to the last PR (can't work in that branch as we consume it in prod)

- renamed variable from `$o-grid-offset-selectors` to `$o-grid-shuffle-selectors`
- decoupled `hide` and `0` selectors

### CODE REVIEW

**Comment by wheresrhys** [comment]
What do you think about the status of center/uncenter too? They're bundled in with human readable classes, but actually provide functionality with no equivalent in the column count based selectors

**Comment by AlbertoElias** [comment]
Pr looks good to me.

Re center, I don't mind if it stays behind the new if clause unless it's widely used, but I've never personally seen it

File: src/scss/_variables.scss [hunk 1]
```diff
@@ -14,6 +14,22 @@
...
+/// @type Bool
+///
+/// Toggle outputting offset, push and pull selectors
+$o-grid-shuffle-selectors: true !default;
+
+/// Human-friendly selectors
+///
+/// @type Bool
+///
+/// Toggle outputting offset, push and pull selectors
```
Comment (kaelig) [inline src/scss/_variables.scss]: The comment seems to refer to something else.

**Comment by kaelig** [comment]
These settings are kind of a footgun. They introduce the possibility that components aren't guaranteed to render well across environments, as grid classes might or might not be available to the browser on certain sites / apps. This could have a detrimental impact on code portability. 

You might want to break the module in two, so that a component can declare it needs push/pull classes by loading an additional module. And you could remove the human friendly classes completey if they are not so useful.

**Comment by alicebartlett** [comment]
@kaelig FWIW I agree with you on the footgun, but Next argued strongly for this fucntionality. We'll be keeping an eye on it.

Origami has some work to do too, we'll be looking at who is looking at the human-friendly classes, and deprecating them if adoption is low enough.

**Comment by wheresrhys** [comment]
Perhaps it needs documenting better, but I see this functionality as being restricted in a similar way to silent mode toggling - for app use only. Components will still have access to mixins, which is all they're supposed to use anyway, and should be resilient to these toggles in the same way they're resilient to silent mode being on/off.

The problem with breaking in two is you then would have two namespaces (unless we introduce the notion of plugins into the origami spec, which has been discussed before).

I have no idea what a footgun is

**Comment by triblondon** [comment]
It assists in the process of shooting oneself in the foot.

**Comment by wheresrhys** [comment]
guns don't shoot feet, people shoot feet

**Pull request merged**